packages/react-vis/src/treemap/index.js

```diff
@@ -91,30 +91,18 @@ function _getScaleFns(props) {
   };
 }
 
-class Treemap extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      scales: _getScaleFns(props),
-      ...getInnerDimensions(props, props.margin)
-    };
-  }
-
-  UNSAFE_componentWillReceiveProps(props) {
-    this.setState({
-      scales: _getScaleFns(props),
-      ...getInnerDimensions(props, props.margin)
-    });
-  }
+function Treemap(props) {
+  const scales = _getScaleFns(props);
+  const innerDimensions = getInnerDimensions(props, props.margin);
 
   /**
    * Create the list of nodes to render.
    * @returns {Array} Array of nodes.
    * @private
    */
-  _getNodesToRender() {
-    const {innerWidth, innerHeight} = this.state;
-    const {data, mode, padding, sortFunction, getSize} = this.props;
+  function _getNodesToRender() {
```

Comment:
Since this function is only used in one place, this should probably be changed into 
```
const nodes = useMemo(() => { ... })
```

Reply:
So I added the `eslint-plugin-react-hooks` because we were missing it, and if I try to wrap the `nodes` computation with `useMemo` I need to have `props` as a dependency: 

> React Hook useMemo has missing dependencies: 'innerDimensions' and 'props'. Either include them or remove the dependency array 

In that case, it would make more sense to me to wrap the component with memo i.e `export default memo(Treemap);` instead of using `useMemo` with props as dependency. wydt?

Reply:
`memo` should work as well since it the funtion depends on the entire `props` object.